test/meta/mtk_structure_indexing.jl

```diff
@@ -0,0 +1,389 @@
+        @test length(plot(sol; idxs = [model.XY,model.Y]).series_list) == 2
+        @test length(plot(sol; idxs = [:X,:Y]).series_list) == 2
+        @test length(plot(sol; idxs = [:XY,:Y]).series_list) == 2
+
+        # As tuple.
+        @test length(plot(sol; idxs = (X, Y)).series_list) == 1
+        @test length(plot(sol; idxs = (XY, Y)).series_list) == 1
+        @test length(plot(sol; idxs = (model.X, model.Y)).series_list) == 1
+        @test length(plot(sol; idxs = (model.XY, model.Y)).series_list) == 1
+        @test length(plot(sol; idxs = (:X, :Y)).series_list) == 1
+        @test length(plot(sol; idxs = (:XY, :Y)).series_list) == 1
+    end     
+end
+
+
+### Tests For Hierarchical System ###
+
+# TODO
+
+### Mass Action Jump Rate Updating Correctness ###
+
+# Checks that the rates of mass action jumps are correctly updated after parameter values are changed.
+let
+    # Creates the model.
+    rn = @reaction_network begin
+        p1*p2, A + B --> C
+    end
+    @unpack p1, p2 = rn
+
+    # Creates a JumpProblem and integrator. Checks that the initial mass action rate is correct.
+    u0 = [:A => 1, :B => 2, :C => 3]
+    ps = [:p1 => 3.0, :p2 => 2.0]
+    dprob = DiscreteProblem(rn, u0, (0.0, 1.0), ps)
+    jprob = JumpProblem(rn, dprob, Direct())
+    jint = init(jprob, SSAStepper())
+    @test jprob.massaction_jump.scaled_rates[1] == 6.0
+
+    # Checks that the mass action rate is correctly updated after normal indexing.
+    jprob.ps[p1] = 4.0
+    @test jprob.massaction_jump.scaled_rates[1] == 8.0
+    jprob.ps[rn.p1] = 5.0
+    @test jprob.massaction_jump.scaled_rates[1] == 10.0
+    jprob.ps[:p1] = 6.0
+    @test jprob.massaction_jump.scaled_rates[1] == 12.0
+    setp(jprob, p1)(jprob, 7.0)
+    @test jprob.massaction_jump.scaled_rates[1] == 14.0
+    setp(jprob, rn.p1)(jprob, 8.0)
+    @test jprob.massaction_jump.scaled_rates[1] == 16.0
+    setp(jprob, :p1)(jprob, 3.0)
+    @test jprob.massaction_jump.scaled_rates[1] == 6.0
+
+    # Check that the mass action rate is correctly updated when `remake` is used.
+    # Checks both when partial and full parameter vectors are provided to `remake`.
+    @test remake(jprob; p = [p1 => 4.0]).massaction_jump.scaled_rates[1] == 8.0
+    @test remake(jprob; p = [rn.p1 => 5.0]).massaction_jump.scaled_rates[1] == 10.0
+    @test remake(jprob; p = [:p1 => 6.0]).massaction_jump.scaled_rates[1] == 12.0
+    @test remake(jprob; p = [p1 => 4.0, p2 => 3.0]).massaction_jump.scaled_rates[1] == 12.0
+    @test remake(jprob; p = [rn.p1 => 5.0, rn.p2 => 4.0]).massaction_jump.scaled_rates[1] == 20.0
+    @test remake(jprob; p = [:p1 => 6.0, :p2 => 5.0]).massaction_jump.scaled_rates[1] == 30.0
+
+    # Checks that updating an integrators parameter values does not affect mass action rate until after
+    # `reset_aggregated_jumps!` have been applied as well (wt which point the correct rate is achieved).
+    jint.ps[p1] = 4.0
+    @test jint.cb.condition.ma_jumps.scaled_rates[1] != 8.0
```

Comment:
It might be a bit better to check the previous value is unchanged, so we know the exact expected behavior is happening.

Reply:
Yes, that makes sense. I will update it to do that.